### ISSUE TITLE
CA1063 and CA1821 produce false positives with expression bodies

### ISSUE THREAD

**Issue by Spaze-Maurice** [dotnet/roslyn-analyzers#1788]
#### Analyzer package
[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.6.1

#### Diagnostic ID

[CA1063](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1063-implement-idisposable-correctly)
[CA1821](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1821-remove-empty-finalizers)

#### Repro steps

This triggers a warning
```csharp
public class SomeTestClass : IDisposable
{
    private readonly Stream resource = new MemoryStream(1024);

    public void Dispose()
    {
        this.Dispose(true);
        GC.SuppressFinalize(this);
    }

    protected virtual void Dispose(bool dispose)
    {
        if (dispose)
        {
            this.resource.Dispose();
        }
    }

    ~SomeTestClass() => this.Dispose(false);
}
```
This does not trigger a warning
```csharp
public class SomeTestClass : IDisposable
{
    private readonly Stream resource = new MemoryStream(1024);

    public void Dispose()
    {
        this.Dispose(true);
        GC.SuppressFinalize(this);
    }

    protected virtual void Dispose(bool dispose)
    {
        if (dispose)
        {
            this.resource.Dispose();
        }
    }

    ~SomeTestClass()
    {
        this.Dispose(false);
    }
}
```

#### Expected behavior
Analyzers works with expression bodies

#### Actual behavior
Analyzers does not work with expression bodies

**Comment by sharwell** [comment]
:+1:

**Comment by mavasani** [comment]
Dupe of https://github.com/dotnet/roslyn-analyzers/issues/1606. Note that the issue was in the underlying compiler/analyzer API, which has been fixed in Visual Studio 15.8 release. Can you try the analyzer package against latest VS 15.8 Preview?

**Comment by Spaze-Maurice** [comment]
I noticed this issue after updating to 15.8. I just installed the latest preview but the issue still persists.

![image](https://user-images.githubusercontent.com/29551709/44284302-7566d100-a261-11e8-8471-e2e53c3adadf.png)

**Comment by mavasani** [comment]
PR: https://github.com/dotnet/roslyn-analyzers/pull/1960
